SquAPI.lua

```diff
@@ -73,1100 +68,1238 @@ squapi.autoFunctionUpdates = true
+        local rot = 10 * bend * (yvel + vel * 10) + headrot[1] * _self.rangeMultiplier
+        local addrot = headrot[2] * _self.rangeMultiplier
+        _self.targets[2] = rot + addrot
+        _self.targets[3] = -rot + addrot
+      else
+        _self.targets[1] = headrot[1] * _self.rangeMultiplier + 2 * bend * (yvel + vel * 15)
+        _self.targets[2] = headrot[2] * _self.rangeMultiplier - svel * 100 * _self.bendStrength
+        _self.targets[3] = _self.targets[2]
+      end
+
+      --ear flicking
+      if _self.doEarFlick then
+        if math.random(0, _self.earFlickChance) == 1 then
+          if math.random(0, 1) == 1 then
+            _self.earx.vel = _self.earx.vel + 50
+          else
+            _self.earz.vel = _self.earz.vel - 50
+          end
         end
+      end
+    else
+      leftEar:setOffsetRot(0, 0, 0)
+      rightEar:setOffsetRot(0, 0, 0)
     end
+  end
 
-    function self:render(dt, context)
-        if self.enabled then
-            self.eary:berp(self.targets[1], dt)
-            self.earx:berp(self.targets[2], dt)
-            self.earz:berp(self.targets[3], dt)
-            
-            local rot3 = self.earx.pos/4
-            local rot3b = self.earz.pos/4
-
-            if self.horizontalEars then
-                local y = self.eary.pos/4
-                self.leftEar:setOffsetRot(y, self.earx.pos/3, rot3)
-                if self.rightEar then 
-                    self.rightEar:setOffsetRot(y, self.earz.pos/3, rot3b) 
-                end
-            else
-                self.leftEar:setOffsetRot(self.eary.pos, rot3, rot3)
-                if self.rightEar then 
-                    self.rightEar:setOffsetRot(self.eary.pos, rot3b, rot3b) 
-                end
-            end
+  ---Run render function on ear
+  ---@param dt number Tick delta
+  function _self:render(dt, _)
+    if _self.enabled then
+      _self.eary:berp(_self.targets[1], dt)
+      _self.earx:berp(_self.targets[2], dt)
+      _self.earz:berp(_self.targets[3], dt)
+
+      local rot3 = _self.earx.pos / 4
+      local rot3b = _self.earz.pos / 4
+
+      if _self.horizontalEars then
+        local y = _self.eary.pos / 4
+        _self.leftEar:setOffsetRot(y, _self.earx.pos / 3, rot3)
+        if _self.rightEar then
+          _self.rightEar:setOffsetRot(y, _self.earz.pos / 3, rot3b)
         end
+      else
+        _self.leftEar:setOffsetRot(_self.eary.pos, rot3, rot3)
+        if _self.rightEar then
+          _self.rightEar:setOffsetRot(_self.eary.pos, rot3b, rot3b)
+        end
+      end
     end
+  end
 
-    table.insert(squapi.ears, self)
-    return self
+  table.insert(squapi.ears, _self)
+  return _self
 end
 
+---CROUCH ANIMATION - This allows you to set an animation for your crouch (this can either be a static pose for crouching, or an animation to transition to crouching)<br><br>It also allows you to optionally set an uncrouch animation, and includes the same features mentioned for crawling if you need as well.
```

Comment:
These two `<br>` tags should be replaced with a markdown paragraph separator (two newlines)

Reply:
Forgor how to make suggestions lol.
```suggestion
---CROUCH ANIMATION - This allows you to set an animation for your crouch (this can either be a static pose for crouching, or an animation to transition to crouching)
---
---It also allows you to optionally set an uncrouch animation, and includes the same features mentioned for crawling if you need as well.
```